python/apps/utils.py

```diff
@@ -2,19 +2,41 @@
 # Licensed under the MIT license.
 
 import numpy as np
+from scipy.cluster.vq import vq, kmeans2
+from typing import Tuple
 
 
-def get_bin_metadata(bin_file):
+def get_bin_metadata(
+    bin_file
+) -> Tuple[int, int]:
     array = np.fromfile(file=bin_file, dtype=np.uint32, count=2)
     return array[0], array[1]
 
 
-def bin_to_numpy(dtype, bin_file):
+def bin_to_numpy(
+    dtype, 
+    bin_file
+) -> np.ndarray:
     npts, ndims = get_bin_metadata(bin_file)
     return np.fromfile(file=bin_file, dtype=dtype, offset=8).reshape(npts, ndims)
 
 
-def read_gt_file(gt_file):
+def numpy_to_bin(array, out_file):
```

Comment:
this actually exists already, check out _builder.py and `from diskannpy import numpy_to_diskann_file`